### PR TITLE
feat: add test infrastructure for the web UI

### LINKED ISSUE #80: Add test infrastructure for the web UI
Also add some simple tests, this will be useful so we can update dependencies using `dependabot` without having some random breakage.

### PR DESCRIPTION
**Pull request by EhabY**
closes #80

### CODE REVIEW

File: ui/tests/components/StatusCardList.test.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    expectedVisible.forEach(name => {
+      expect(screen.queryByText(name)).toBeInTheDocument()
+    })
+
+    expectedHidden.forEach(name => {
+      expect(screen.queryByText(name)).not.toBeInTheDocument()
+    })
+  })
+})
```
Comment (yazan-abu-obaideh) [inline ui/tests/components/StatusCardList.test.tsx:49]: Very nice

File: ui/src/components/ColorToggleButton.tsx [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { MoonIcon, SunIcon } from "@chakra-ui/icons"
 import { IconButton, IconButtonProps, useColorMode } from "@chakra-ui/react"
 import React from "react"
 
-function ColorToggleButton(props: Partial<IconButtonProps>): React.ReactElement {
+function ColorToggleButton(props: Readonly<Partial<IconButtonProps>>): React.ReactElement {
```
Comment (yazan-abu-obaideh) [inline ui/src/components/ColorToggleButton.tsx:5]: Why `Readonly`?

File: ui/tests/components/StatusCardList.test.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    for (const status of statuses) {
+      expect(screen.getByText(status.Name)).toBeInTheDocument()
+    }
+  })
+
+  it("should filter out the whole list of statuses if query is not found", () => {
+    render(<StatusCardList statuses={statuses} query="NOT FOUND!!!" />)
+    expect(screen.queryByText("Foobar")).not.toBeInTheDocument()
+    expect(screen.queryByText("FooFoo")).not.toBeInTheDocument()
+    expect(screen.queryByText("Boobaz")).not.toBeInTheDocument()
```
Comment (yazan-abu-obaideh) [inline ui/tests/components/StatusCardList.test.tsx:28]: Could be a loop for brevity and consistency

File: ui/tests/components/StatusCardList.test.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    expect(screen.getByText(/No items/i)).toBeInTheDocument()
+  })
+
+  it.each([
+    { query: "Foo", expectedVisible: ["Foobar", "FooFoo"], expectedHidden: ["Boobaz"] },
+    { query: "Foob", expectedVisible: ["Foobar"], expectedHidden: ["FooFoo", "Boobaz"] },
```
Comment (yazan-abu-obaideh) [inline ui/tests/components/StatusCardList.test.tsx:34]: I'd also like to see a scenario for 'oob' for example (do we expect to see "Foobar"?)

**Comment by yazan-abu-obaideh** [comment]
@m-adawi @EhabY Do you guys think a ticket should be created so these tests are run in the pipeline?

File: ui/src/components/ColorToggleButton.tsx (hunk 2 above)
Comment (EhabY) [reply ui/src/components/ColorToggleButton.tsx:5]: Sonarlint, this is just a precaution, arguments here should never be modified

File: ui/tests/components/StatusCardList.test.tsx (hunk 3 above)
Comment (EhabY) [reply ui/tests/components/StatusCardList.test.tsx:28]: Done

File: ui/tests/components/StatusCardList.test.tsx (hunk 4 above)
Comment (EhabY) [reply ui/tests/components/StatusCardList.test.tsx:34]: Ah smart, added that test

File: ui/tests/components/StatusCardList.test.tsx (hunk 1 above)
Comment (EhabY) [reply ui/tests/components/StatusCardList.test.tsx:49]: <3

File: Dockerfile [hunk 5]
```diff
@@ -3,7 +3,7 @@ WORKDIR /ui
 COPY ui/package.json ui/package-lock.json .
 RUN npm install
 COPY ui/ .
-RUN npm run build
```
Comment (EhabY) [inline Dockerfile:6]: Is this needed @m-adawi ?

**Review by m-adawi — APPROVED** [review]
LGTM

**Pull request merged**